pkg/dns/client/dns.go

```diff
@@ -187,11 +208,17 @@ func (h *LocalDNSServer) ServeDNS(proxy *dnsProxy, w dns.ResponseWriter, req *dn
 	lp := h.lookupTable.Load()
 	hostname := strings.ToLower(req.Question[0].Name)
 	if lp == nil {
-		log.Debugf("dns request for host %q before lookup table is loaded", hostname)
-		response = new(dns.Msg)
-		response.SetReply(req)
-		response.Rcode = dns.RcodeServerFailure
-		_ = w.WriteMsg(response)
+		if os.Getuid() != 0 {
```

Comment:
nit: Getuid makes a full syscall, seems like we can just check once

Reply:
And it seems like this should be tied to using /var/lib/istio/resolv.conf as well?

Reply:
In the big scheme of startup thing - a syscall is the last thing to optimize :-)

Running as UID=0 with iptables is pretty bad for envoy - will loop. We can add an extra env or flag - but if we start at root 
and the flag is not set, bad things can happen ( not for DNS, but for envoy exec ).

I'm a bit reluctant to add even more flags and envs - I think relying on the environment works better, less changes
to miss-configure. I think running as UID 0 is pretty clear user intent.